plugins/inputs/icinga2/icinga2.go

```diff
@@ -0,0 +1,154 @@
+package icinga2
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Icinga2 struct {
+	Server          string
+	Filter          string
+	Username        string
+	Password        string
+	ResponseTimeout internal.Duration
+	tls.ClientConfig
+
+	client *http.Client
+}
+
+type Result struct {
+	Results []Object `json:"results"`
+}
+
+type Object struct {
+	Attrs Attribute  `json:"attrs"`
+	Name  string     `json:"name"`
+	Joins struct{}   `json:"joins"`
+	Meta  struct{}   `json:"meta"`
+	Type  ObjectType `json:"type"`
+}
+
+type Attribute struct {
+	CheckCommand string `json:"check_command"`
+	DisplayName  string `json:"display_name"`
+	Name         string `json:"name"`
+	State        int    `json:"state"`
+}
+
+type ObjectType string
+
+var sampleConfig = `
+  ## Required Icinga2 server address (default: "https://localhost:5665")
+  # server = "https://localhost:5665"
+  
+  ## Required Icinga2 object type ("services" or "hosts, default "services")
+  # filter = "services"
```

Comment:
Maybe make this a bool, something like defaulting to `hosts=false`, meaning it will collect services only; unless there are other object types besides `services` and `hosts`

Reply:
I actually prefer this style, if we have hosts=false then it isn't clear that it means services.  Also if another object type has support added then a bool won't scale up to additional items.  I also wonder if a user would ever want to collect both services and hosts?  Maybe this should be an array: `["services", "hosts"]`?  Finally, filter is a somewhat confusing because normally you filter things out, maybe it should be renamed `object_type`?

Reply:
Oh ya, good call

Reply:
I agree, i renamed to `object_type`